machetli/environments.py

```diff
@@ -63,41 +80,38 @@ def __init__(self, allow_nondeterministic_successor_choice=True,
         self.allow_nondeterministic_successor_choice = \
             allow_nondeterministic_successor_choice
 
-    def submit(self, batch, evaluator_path):
+    def run(self, evaluator_path, batch, on_task_completed):
```

Comment:
We expect this function to have a return value in the search, so we should proably ensure this is the case.
```suggestion
    def run(self, evaluator_path, batch, on_task_completed) -> list[EvaluationTask]:
```